notebooks/Update_TCGA_Data_Model_UUID_to_DRS_Identifier.ipynb

```diff
@@ -0,0 +1,239 @@
+    "    df = pd.read_csv(StringIO(res_etype.text), sep=\"\\t\")\n",
+    "    original_tsv_name = \"original_\" + etype + \"_table.tsv\"\n",
+    "    # write data frames to .tsv files\n",
+    "    df.to_csv(original_tsv_name, sep=\"\\t\", index=False)\n",
+    "    !gsutil cp $original_tsv_name $ws_bucket 2> stdout\n",
+    "    \n",
+    "    # read entity table response into dictionary to perform DRS URL updates \n",
+    "    dict_etype = list(csv.DictReader(StringIO(res_etype.text), delimiter='\\t'))\n",
+    "\n",
+    "    \n",
+    "    # create empty list to add updated rows and list to capture list of columns that were modified\n",
+    "    drs_dict_table = []\n",
+    "    modified_cols = []\n",
+    "    # for \"row\" (each row is [list] of column:values)\n",
+    "    for row in dict_etype:\n",
+    "        drs_row = row.copy()\n",
+    "        \n",
+    "        # for each column in row\n",
+    "        for col in row:\n",
+    "            guid = row[col].split(\"/\")[0]\n",
+    "            # check if guid exists and is valud guid in all columns but ID column\n",
+    "            if (col.endswith(col_suffix)) and (guid and guid_pattern.match(guid)):\n",
+    "                drs_url = \"drs://dg.4DFC:\" + guid\n",
+    "                drs_row[col] = drs_url\n",
+    "                modified_cols.append(col)\n",
+    "            else:\n",
+    "                None\n",
+    "\n",
+    "        # append new \"row\" with updated drs values to new list\n",
+    "        drs_dict_table.append(drs_row)\n",
+    "        \n",
+    "        # set output file name and write tsv files\n",
+    "        updated_tsv_name = \"updated_\" + etype + \"_table.tsv\"\n",
+    "        tsv_headers = drs_dict_table[0].keys()\n",
+    "        \n",
+    "        with open(updated_tsv_name, 'w') as outfile:\n",
+    "            # get keys from OrderedDictionary and write rows, separate with tabs\n",
+    "            writer = csv.DictWriter(outfile, tsv_headers, delimiter=\"\\t\")\n",
+    "            writer.writeheader()\n",
+    "            writer.writerows(drs_dict_table)\n",
+    "    \n",
+    "    print(f'Saving DRS URL updated {etype} TSV to {ws_bucket}')\n",
+    "    !gsutil cp $updated_tsv_name $ws_bucket 2> stdout\n",
+    "    \n",
+    "    modified_cols = list(set(modified_cols))\n",
+    "    if dry_run:\n",
+    "        print(f'The following columns in the {etype} table *will be* be updated:')\n",
+    "        if not modified_cols:\n",
+    "            print('\\t' * 4 + f\"No columns to update in the {etype} table.\" + \"\\n\\n\")\n",
+    "        else:\n",
+    "            print('\\n'.join(['\\t' * 4 + c for c in modified_cols]))\n",
+    "            print(f'To view what will be updated, inspect the {updated_tsv_name} file in the workspace bucket, {ws_bucket}.' + \"\\n\\n\")\n",
+    "    else:\n",
+    "        # upload newly created tsv file containing drs urls\n",
```

Comment:
maybe add a print statement saying something like "Updating entities..." in case this API call takes a few seconds